added documentation for the gravity compensation package

Shiming-Liang opened the pull request:
- added documentation for the gravity compensation package
- changed the requirement.txt to fix pip dependency issues

Shiming-Liang committed:
added documentation for the gravity compensation package

Shiming-Liang committed:
Updated doc for the gravity compensation package

Added a more lively overview

lukeschmitt-tr commented on docs/ros2_packages/gravity_compensation.rst:
```diff
@@ -0,0 +1,160 @@
+========
+
+As its name suggests, gravity compensation cancels out the effect of the gravity on a system.
+In our case, we compensate the gravity on an interbotix arm with its joint motors.
```
```suggestion
In our case, we compensate the gravity on an Interbotix arm with its joint motors.
```

lukeschmitt-tr commented on docs/ros2_packages/gravity_compensation.rst:
```diff
@@ -0,0 +1,160 @@
+In our case, we compensate the gravity on an interbotix arm with its joint motors.
+This feature is useful when we use the arm as a teleoperation remote or when we teach it a specific trajectory and don't want to keep holding the arm to avoid it from collapsing.
+
+.. note:: Currently supported arms include: WidowX-250 6DOF and ALOHA WidowX-250 6DOF.
```
Style admonitions like this

```suggestion
.. note::

    Currently supported arms include: WidowX-250 6DOF and ALOHA WidowX-250 6DOF.
```

lukeschmitt-tr commented on docs/ros2_packages/gravity_compensation.rst:
```diff
@@ -0,0 +1,160 @@
+        static_friction_coefficient: 0.0
+        dither_speed: 0.0
+
+.. warning:: Excessive dithering WILL cause heat and wear on the joints. Please use it with caution.
```
Once admonition style is fixed, make sure to update this such that it's one sentence per line.

Shiming-Liang committed:
Update docs/ros2_packages/gravity_compensation.rst

fixed a typo

Co-authored-by: lukeschmitt-tr <85308904+lukeschmitt-tr@users.noreply.github.com>

Shiming-Liang committed:
modified style admonitions for the gravity compensation page

lukeschmitt-tr reviewed (approved):
Approved, wait to merge until the toolboxes and manipulators PRs are merged.

Shiming-Liang committed:
Update on the gravity compensation doc

- removed mode configs
- changed namespace to robot_name to be more informative
- fixed ' ' pairs in the csv file
